plugin/trino-faker/src/main/java/io/trino/plugin/faker/Literal.java

```diff
@@ -0,0 +1,247 @@
+            return parseBoolean(value);
+        }
+        if (DATE.equals(type)) {
+            return (long) parseDate(value);
+        }
+        if (type instanceof DecimalType decimalType) {
+            return parseDecimal(value, decimalType);
+        }
+        if (REAL.equals(type)) {
+            return (long) floatToRawIntBits(Float.parseFloat(value));
+        }
+        if (DOUBLE.equals(type)) {
+            return Double.parseDouble(value);
+        }
+        // not supported: HYPER_LOG_LOG, QDIGEST, TDIGEST, P4_HYPER_LOG_LOG
+        if (INTERVAL_DAY_TIME.equals(type)) {
+            return parseDayTimeInterval(value, IntervalLiteral.IntervalField.SECOND, Optional.empty());
+        }
+        if (INTERVAL_YEAR_MONTH.equals(type)) {
+            return parseYearMonthInterval(value, IntervalLiteral.IntervalField.MONTH, Optional.empty());
+        }
+        if (type instanceof TimestampType timestampType) {
+            return parseTimestamp(timestampType.getPrecision(), value);
+        }
+        if (type instanceof TimestampWithTimeZoneType timestampWithTimeZoneType) {
+            return parseTimestampWithTimeZone(timestampWithTimeZoneType.getPrecision(), value);
+        }
+        if (type instanceof TimeType) {
+            return parseTime(value);
+        }
+        if (type instanceof TimeWithTimeZoneType timeWithTimeZoneType) {
+            return parseTimeWithTimeZone(timeWithTimeZoneType.getPrecision(), value);
+        }
+        if (VARBINARY.equals(type)) {
+            return parseBinary(value);
+        }
+        if (type instanceof VarcharType || type instanceof CharType) {
+            return utf8Slice(value);
+        }
+        // not supported: ROW, ARRAY, MAP, JSON
+        if (type instanceof IpAddressType) {
+            return parseIpAddress(value);
+        }
+        // not supported: GEOMETRY
+        if (type instanceof UuidType) {
+            return javaUuidToTrinoUuid(UUID.fromString(value));
+        }
+        throw new IllegalArgumentException("Unsupported literal type: " + type);
+    }
+
+    private static long parseLong(String value)
+    {
+        value = value.replace("_", "");
+
+        if (value.startsWith("0x") || value.startsWith("0X")) {
+            return Long.parseLong(value.substring(2), 16);
+        }
+        else if (value.startsWith("0b") || value.startsWith("0B")) {
+            return Long.parseLong(value.substring(2), 2);
+        }
+        else if (value.startsWith("0o") || value.startsWith("0O")) {
+            return Long.parseLong(value.substring(2), 8);
+        }
+        else {
+            return Long.parseLong(value);
+        }
+    }
+
+    private static Boolean parseBoolean(String value)
+    {
+        value = value.toLowerCase(ENGLISH);
+        checkArgument(value.equals("true") || value.equals("false"));
+        return value.equals("true");
+    }
+
+    private static Object parseDecimal(String value, DecimalType decimalType)
```

Comment:
No, because it infers the data type from the value, and can sometimes return a short decimal when a long decimal is expected. I added a code comment to mention this.